### PR TITLE
Fix: drop spans in BeforeSpanCallback.

### LINKED ISSUE #1708: BeforeSpanCallback isnt dropping spans
Bug introduced by https://github.com/getsentry/sentry-java/pull/1699 since the span isn't finished, but the integration does it automatically

### PR DESCRIPTION
**Pull request by maciejwalkowiak**
## :scroll: Description

Fix: drop spans in BeforeSpanCallback. #1713

## :bulb: Motivation and Context

Fixes #1708

## :pencil: Checklist

- [x] I reviewed the submitted code
- [x] I added tests to verify the changes
- [ ] I updated the docs if needed
- [x] No breaking changes

_#skip-changelog_

### CODE REVIEW

File: sentry-openfeign/src/main/java/io/sentry/openfeign/SentryFeignClient.java [hunk 1]
```diff
@@ -60,9 +60,16 @@ public Response execute(final @NotNull Request request, final @NotNull Request.O
...
+          final ISpan result = beforeSpan.execute(span, request, response);
+
+          if (result == null) {
+            // span is dropped
+            span.getSpanContext().setSampled(false);
+          } else {
+            span = result;
+            span.finish();
+          }
+        } else {
```
Comment (marandaneto) [inline sentry-openfeign/src/main/java/io/sentry/openfeign/SentryFeignClient.java:72]: the logic here is a bit different than the one in the OkHttp, which one is correct?

Comment (marandaneto) [reply sentry-openfeign/src/main/java/io/sentry/openfeign/SentryFeignClient.java:72]: if I got it right, we dont finish the span in case its dropped

Comment (maciejwalkowiak) [reply sentry-openfeign/src/main/java/io/sentry/openfeign/SentryFeignClient.java]: Thanks, good catch.

**Review by marandaneto — APPROVED** [review]

**Comment by marandaneto** [comment]
this one should have a changelog since beta.1 has been published, adding it

**Pull request merged**